src/dds/rsdds-device-factory.cpp

```diff
@@ -163,6 +172,43 @@ std::vector< std::shared_ptr< device_info > > rsdds_device_factory::query_device
     {
         unsigned const mask = context::combine_device_masks( requested_mask, get_context()->get_device_mask() );
 
+        auto participant = _watcher_singleton->get_device_watcher()->get_participant();
+        domain_context * p_domain = nullptr;
+        {
+            std::lock_guard< std::mutex > lock( domain_context_by_id_mutex );
+            auto it = domain_context_by_id.find( participant->domain_id() );
+            if( it != domain_context_by_id.end() )
+                p_domain = &it->second;
+        }
+        if( p_domain )
+        {
+            std::lock_guard< std::mutex > lock( p_domain->wait_mutex );
+            if( p_domain->seconds_to_wait > 0 )
+            {
+                // do this within the mutex: if multiple threads all try to query_devices, the others will
```

Comment:
The others will what?
A line was deleted?

Reply:
Yes:
```
                // until we're done
```
Added.